physlr/physlr.py

```diff
@@ -1332,6 +1332,79 @@ def determine_molecules_bc_cosine_of_squared(g, u):
                 for community in
                 Physlr.detect_communities_cosine_of_squared(g, bi_connected_component)]
 
+    @staticmethod
+    def partition_subgraph_into_bins_randomly(node_set, max_size=40):
+        """Prtition the subgraph into bins randomly for faster processing. Return bins."""
+        bins_count = 1
+        if len(node_set) > max_size:
+            bins_count = 1 + int(len(node_set) / max_size)
+        node_list = list(node_set)
+        random.shuffle(node_list)
+        size, leftover = divmod(len(node_set), bins_count)
+        bins = [node_list[0 + size * i: size * (i + 1)] for i in list(range(bins_count))]
+        edge = size * bins_count
+        for i in list(range(leftover)):
```

Comment:
```suggestion
        for i in range(leftover):
```
Do not convert to `list` where it's not necessary.